clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -195,8 +204,11 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
         // If non-empty these records should be returned before incrementing the reader.
         // This is used for read-ahead and inserting synthetic records.
         // We use a deque so we can iterate over it.
-        std::deque<RecordType> queue;
+        // Remember to adjust real_records_in_queue when adding or removing from this.
+        std::deque<cached_record_t> queue;
+        uint64_t real_records_in_queue = 0;
         bool cur_from_queue;
+        bool is_cur_record_real;
```

Comment:
Add an initial value.  Ditto for line 210 (I think it's set in the code before read so no current bug I don't think, but safest to have an initial value).